content/blog/2019-12-18-polynomial/index.md

```diff
@@ -0,0 +1,155 @@
+multiplication for this kind of crypto systems. 
+
+## Motivation
+
+### Post-Quntum Cryptography
+
+A lot of public-key cryptography systems we used today relying on primitives
+such as RSA and elliptic curve which are vulnerable under the quantum computer
+attacks.  As people now suggest that the threat of the quantum computer might
+not be that far in the future, cryptography researchers have been studying
+so-called post-quantum cryptography (PQC), the kind of crypto primitives that we
+believe can withstand the quantum attack.  There are a lot of post-quantum
+cryptosystems being proposed recently and post-quantum cryptography researchers
+now are actively trying to analyze the security and performance of these
+systems.
+
+### PQC Candidates
+
+Although the question about how far are we from building a quantum computer that
+can break the secure communication today is still in heated debate, we now need
+a guideline for people who want to migrate to choose among all those
+post-quantum crypto systems proposed in the literature so far.  National
+Institute of Standards and Technology (NIST) started a
+[process](https://csrc.nist.gov/Projects/post-quantum-cryptography) to standardize
+the post-quantum in 2017 and as of today (Dec 2019), there are still 17
+candidates for public-key encryption and 9 candidates for digital signature in
+[round 2](https://csrc.nist.gov/Projects/post-quantum-cryptography/round-2-submissions).
+When choosing which systems to become the standard, security is obviously the
+priority. However, the performance the systems is also very important because
+once chosen as part of the standard, it would very likely to be implemented in
+popular library such as openssl and deployed to millions of devices and servers
+and have impact on billions of users.
+
+### High Performance PQC Implementation 
+
+Thus, it is very important to anaylize the performance of these systems, and
+know that how the most opimized version of the code perform on each platforms.
+Yet, turning an algorithm into highly optimizing code is not an easy job. To
+make things even harder, now there are many different crypto systems and each of
+them have different level of security parameters to select.  Hand crafting
+optimizing code for every one of them and analyze their performance is a lot of
+work and it is also not a fair comparison among different systems if some
+researchers just don't have the resource to invest a lot of time on crafting
+those highly optimized assembly code. Here, we hope to start addressing this
+problem by creating an code generating system where its goal is to create high
+performance polynomial multiplication code commonly used in a lattice-based
+post-quantum crpyto systems.
+
+### High Performance Code Generation
+
+Generating high performance code to run on different hardware have been studied
+for decades. Signal processing community have been studied how to generate high
```

Comment:
- The signal processing community
- have been studied -> has studied (check the subject of this verb)